dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -283,14 +368,26 @@ public CachedVanityUrl getLiveCachedVanityUrl(final String uri, final Host site,
      * @param uri The current uri
      * @return true if the CachedVanityUrl match, false if not
      */
-    private boolean patternMatches(CachedVanityUrl cachedVanityUrl, String uri) {
-        boolean patternMatches = false;
+    private VanityMatches patternMatches(final CachedVanityUrl cachedVanityUrl,
+                                         final String uri) {
         if (cachedVanityUrl != null) {
-            Matcher matcher = cachedVanityUrl.getPattern().matcher(uri);
-            patternMatches = matcher.matches();
+
+            final Matcher matcher = cachedVanityUrl.getPattern().matcher(uri);
+            if (matcher.matches()) {
+                if (matcher.groupCount() > 0) {
+
+                    final String[] matches = new String[matcher.groupCount()];
+                    rangeClosed(1, matcher.groupCount())
+                            .forEach(i -> matches[i - 1] = matcher.group(i));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'i' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)